include/mneme/view.hpp

```diff
@@ -1,23 +1,34 @@
 #ifndef MNEME_VIEW_H_
 #define MNEME_VIEW_H_
 
+#include "iterator.hpp"
+#include "span.hpp"
+
 #include <cstddef>
 #include <sstream>
 #include <stdexcept>
 #include <vector>
 
-#include "iterator.hpp"
-
 namespace mneme {
 
 template <typename Storage, std::size_t Stride = dynamic_extent> class StridedView {
 public:
     using offset_type = typename Storage::offset_type;
     using iterator = Iterator<StridedView<Storage, Stride>>;
 
+    StridedView() : size_(0), container(nullptr) {}
+
     template <class Layout>
-    StridedView(Layout const& layout, Storage& container, std::size_t from, std::size_t to)
-        : size_(to - from), container(container), offset(container.offset(layout[from])) {
+    StridedView(Layout const& layout, Storage& container, std::size_t from, std::size_t to) {
+        setStorage(layout, container, from, to);
+    }
+
+    template <class Layout>
+    void setStorage(Layout const& layout, Storage& cntnr, std::size_t from, std::size_t to) {
```

Comment:
Can you please rename this to container? Took me some time to figure out what cntnr means. Same below as well.